src/main/java/Mariogatchi/Inventory.java

```diff
@@ -1,4 +1,89 @@
 package Mariogatchi;
 
-public class Inventory {
+import java.io.Serializable;
+import java.util.Map;
+
+
+
+public class Inventory implements Serializable {
+    private Map<Item, Integer> itemToQuantity;
+
+    // max capacity of the mariogatchi.Inventory. The sum of quantities of each item in itemToQuantity should not exceed this amount
+    private int capacity;
+
+
+    // the amount of space occupied in this inventory. The sum of quantities of each item in itemToQuantity.
+    private int occupied;
+
+
+    public Inventory(Map<Item, Integer> defaultItemToQuantity) {
+        this.itemToQuantity = defaultItemToQuantity;
+    }
+
+
+    /*
+    Decrement the quantity of item in itemToQuantity by the given quantityToRemove.
+    Return true if this is successfully done, return false otherwise.
+     */
+    public boolean removeItem(Item item, int quantityToRemove) {
+        if (itemToQuantity.containsKey(item) && quantityToRemove <= itemToQuantity.get(item)){
+            int newQuantity = itemToQuantity.get(item) - quantityToRemove;
+            itemToQuantity.replace(item, newQuantity);
```

Comment:
Will this leave ghost items in the inventory if the newQuantity is zero?

Reply:
It will now remove the item if the new quantity is 0